core/node/db_pruner/src/tests.rs

```diff
@@ -324,3 +355,59 @@ async fn pruner_is_resistant_to_errors() {
     stop_sender.send_replace(true);
     pruner_task_handle.await.unwrap().unwrap();
 }
+
+#[tokio::test]
+async fn pruning_iteration_timely_shuts_down() {
+    let pool = ConnectionPool::<Core>::test_pool().await;
+    let mut conn = pool.connection().await.unwrap();
+    insert_l2_blocks(&mut conn, 10, 2).await;
+
+    let pruner = DbPruner::with_conditions(
+        DbPrunerConfig {
+            removal_delay: Duration::MAX, // intentionally chosen so that pruning iterations stuck
+            pruned_batch_chunk_size: 3,
+            minimum_l1_batch_age: Duration::ZERO,
+        },
+        pool.clone(),
+        vec![], //No checks, so every batch is prunable
```

Comment:
Well, now things got serious. In the first comment, you should start with a capital letter, and in the second one you skipped a space between `//` and `N`. ❗ 😠 ❗ 

(jk)